### PR TITLE
[Merged by Bors] - refactor: Deduplicate monotonicity of `•` lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Remove the duplicates introduced in #8869 by sorting the lemmas in `Algebra.Order.SMul` into three files:
* `Algebra.Order.Module.Defs` for the order isomorphism induced by scalar multiplication by a positivity element
* `Algebra.Order.Module.Pointwise` for the order properties of scalar multiplication of sets. This file is new. I credit myself for https://github.com/leanprover-community/mathlib/pull/9078
* `Algebra.Order.Module.OrderedSMul`: The material about `OrderedSMul` per se. Inherits the copyright header from `Algebra.Order.SMul`. This file should eventually be deleted.

I move each `#align` to the correct file. On top of that, I delete unused redundant `OrderedSMul` instances (they were useful in Lean 3, but not anymore) and `eq_of_smul_eq_smul_of_pos_of_le`/`eq_of_smul_eq_smul_of_neg_of_le` since those lemmas are weird and unused.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+## Implementation notes
+* We choose to define `OrderedSMul` as a `Prop`-valued mixin, so that it can be
+  used for actions, modules, and algebras
+  (the axioms for an "ordered algebra" are exactly that the algebra is ordered as a module).
+* To get ordered modules and ordered vector spaces, it suffices to replace the
+  `OrderedAddCommMonoid` and the `OrderedSemiring` as desired.
+
+## TODO
+
+This file is now mostly useless. We should try deleting `OrderedSMul`
```
Comment (jcommelin) [inline Mathlib/Algebra/Order/Module/OrderedSMul.lean:36]: What is it superseded by?

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Module/OrderedSMul.lean:36]: The `PosSMulMono` and friends mixins. `OrderedSMul` is a mixin that effectively combines`PosSMulStrictMono` and `PosSMulReflectLT` together.

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bump

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**